Dockerfile.debug-init

```diff
@@ -0,0 +1,37 @@
+FROM public.ecr.aws/amazonlinux/amazonlinux:latest as init-builder
+
+RUN curl -sL -o /bin/gimme https://raw.githubusercontent.com/travis-ci/gimme/master/gimme
+RUN chmod +x /bin/gimme
+RUN yum upgrade -y && yum install -y tar gzip git
+ENV HOME /home
+
+# Lock Go Lang version to stable
+RUN export GO_STABLE_VERSION=`curl --silent https://go.dev/VERSION?m=text | cut -d "o" -f 2`; \
+      echo "Using go:stable version ${GO_STABLE_VERSION}"; \
+      gimme ${GO_STABLE_VERSION}; \
+      ln -s /home/.gimme/versions/go${GO_STABLE_VERSION}.linux.arm64 /home/.gimme/versions/gostable.linux.arm64; \
+      ln -s /home/.gimme/versions/go${GO_STABLE_VERSION}.linux.amd64 /home/.gimme/versions/gostable.linux.amd64
+ENV PATH ${PATH}:/home/.gimme/versions/gostable.linux.arm64/bin:/home/.gimme/versions/gostable.linux.amd64/bin
+RUN go version
+
+ENV GO111MODULE on
+RUN go env -w GOPROXY=direct
+
+# Build init process for Fluent Bit
+COPY /init/fluent_bit_init_process.go /
+COPY /go.mod /
+COPY /go.sum /
+RUN go mod tidy \
+    && go build fluent_bit_init_process.go
+
+FROM amazon/aws-for-fluent-bit:debug
```

Comment:
This is the only different line

Reply:
@matthewfala I started working on this a while ago but stopped because it was a bit more complicated and time consuming than I had hoped. See the init controls the fluent bit command, so if we truly want a debug dockerfile with a core dump and a valgrind option we need a change in the init process as well. 

I guess its fine not to do that right now... having core dump debug init build is a good first step. I wonder if we can re-use the same dockerfile if this is the only line changed- could we use docker args to determine which is the base image? Similar to what we have in our Dockerfile.plugins: https://github.com/aws/aws-for-fluent-bit/blob/mainline/Dockerfile.plugins#L21

Reply:
Here was the previous work I tried to do: https://github.com/PettitWesley/aws-for-fluent-bit/commit/1c1e52e82d789d2879d2288baa8d72167079271b